src/libraries/Common/tests/System/Security/Cryptography/X509Certificates/X509CertificateLoaderTests.cs

```diff
@@ -0,0 +1,284 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    public class X509CertificateLoaderTests_FromByteArray : X509CertificateLoaderTests
+    {
+        protected override void NullInputAssert(Action action) =>
+            AssertExtensions.Throws<ArgumentNullException>("data", action);
+
+        protected override void EmptyInputAssert(Action action) =>
+            Assert.ThrowsAny<CryptographicException>(action);
+
+        protected override X509Certificate2 LoadCertificate(byte[] bytes, string path) =>
+            X509CertificateLoader.LoadCertificate(bytes);
+
+        protected override X509Certificate2 LoadCertificateFileOnly(string path) =>
+            X509CertificateLoader.LoadCertificate(File.ReadAllBytes(path));
+
+        protected override bool TryGetContentType(byte[] bytes, string path, out X509ContentType contentType)
+        {
+            if (bytes is null)
+            {
+                contentType = X509ContentType.Unknown;
+                return false;
+            }
+
+            contentType = X509Certificate2.GetCertContentType(bytes);
+            return true;
+        }
+    }
+
+    public class X509CertificateLoaderTests_FromByteSpan : X509CertificateLoaderTests
+    {
+        protected override void NullInputAssert(Action action) =>
+            Assert.ThrowsAny<CryptographicException>(action);
+
+        protected override void EmptyInputAssert(Action action) =>
+            Assert.ThrowsAny<CryptographicException>(action);
+
+        protected override X509Certificate2 LoadCertificate(byte[] bytes, string path) =>
+            X509CertificateLoader.LoadCertificate(new ReadOnlySpan<byte>(bytes));
+
+        protected override X509Certificate2 LoadCertificateAtOffset(byte[] bytes, int offset) =>
+            X509CertificateLoader.LoadCertificate(bytes.AsSpan(offset));
+
+        protected override X509Certificate2 LoadCertificateFileOnly(string path)
+        {
+            // Use a strategy other than File.ReadAllBytes.
+            using (FileStream stream = File.OpenRead(path))
+            using (MemoryManager<byte> manager = MemoryMappedFileMemoryManager.CreateFromFileClamped(stream))
+            {
+                return X509CertificateLoader.LoadCertificate(manager.Memory.Span);
+            }
+        }
+
+        protected override bool TryGetContentType(byte[] bytes, string path, out X509ContentType contentType)
+        {
+            contentType = X509ContentType.Unknown;
```

Comment:
Why does the span implementation always return unknown while the array one does a live class to `GetCertContentType`?

Reply:
All of the data in the tests ultimately comes from a file, or a byte[] (or each independently), so the span version is either seeing redundant data from byte[] or has a Memory-Mapped view of the file.

Combined with GetCertContentType not being available on all TFMs (which matters for the backcompat package) it seemed not worth the hassle, given that this is only ever used as a precondition.

(To which you reply: "Cool.  Add a comment.")